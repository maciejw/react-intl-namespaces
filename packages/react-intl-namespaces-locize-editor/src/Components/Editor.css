body.searchEnabled * {
  pointer-events: auto;
}
