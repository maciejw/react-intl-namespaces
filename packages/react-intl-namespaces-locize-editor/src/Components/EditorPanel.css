.button {
  border: none;
  color: #fff;
  cursor: pointer;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 14px;
  font-weight: 300;
  height: 30px;
  line-height: 30px;
  min-width: 90px;
  outline: none;
  padding: 0 2px;
  text-align: center;
  text-decoration: none;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
}
.button:hover {
  opacity: 0.9;
}
.button:active {
  padding: 0 1px 0 3px;
}
.buttonOff {
  background-color: #d50000;
}
.buttonOn {
  background-color: #54a229;
}
.option {
  text-transform: uppercase;
}
.panel {
  background-color: #fff;
  border: solid 1px #1976d2;
  bottom: 2px;
  box-shadow: 0px 1px 2px 0px rgba(0, 0, 0, 0.5);
  font-family: Helvetica, Arial, sans-serif;
  padding: 10px;
  position: fixed;
  right: 18px;
  z-index: 2001;
  user-select: none;
}
.panel.pinned {
  max-width: 1000px;
}
.panel .content {
  display: none;
  float: left;
}
.panel.pinned .content {
  display: block;
  width: auto;
}
.select {
  width: 50px;
  border: none;
  box-sizing: content-box;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 14px;
  font-weight: 300;
  padding: 6px;
  text-transform: uppercase;
}

.title {
  color: #1976d2;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 14px;
  font-weight: 300;
  margin: 0 0 5px 0;
}

.pin {
  height: 52px;
  min-width: 20px;
  float: right;
  margin-left: 5px;
}
.pin:active {
  padding: initial;
}
.pin::after {
  font-size: 30px;
}
.pin.pinned::after {
  content: '\203A';
}
.pin.unpinned::after {
  content: '\2039';
}
