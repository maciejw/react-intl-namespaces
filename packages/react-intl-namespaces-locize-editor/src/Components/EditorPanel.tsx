import classNames from 'classnames';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import styles from './EditorPanel.css';

export class EditorPanel extends React.Component<
  EditorPanel.Props,
  EditorPanel.State
> {
  constructor(props: EditorPanel.Props, state: EditorPanel.State) {
    super(props, state);
  }

  public render() {
    const languages = this.props.languages;
    return (
      <div
        className={classNames(styles.panel, {
          [styles.pinned]: this.props.pinned,
        })}
      >
        {
          <span className={classNames(styles.content)}>
            <div className={styles.title}>locize editor</div>
            {languages.length > 0 && (
              <select
                className={styles.select}
                value={this.props.language}
                onChange={e => this.props.onChangeLanguage(e.target.value)}
              >
                {languages.map(l => (
                  <option key={l} value={l} className={styles.option}>
                    {l}
                  </option>
                ))}
              </select>
            )}{' '}
            <button
              onClick={() => this.props.onToggleSearch()}
              className={classNames(styles.button, {
                [styles.buttonOn]: this.props.searchEnabled,
                [styles.buttonOff]: !this.props.searchEnabled,
              })}
            >
              {this.props.searchEnabled ? 'On' : 'Off'}
            </button>{' '}
            <button
              onClick={() => this.props.onShowIds()}
              className={classNames(styles.button, {
                [styles.buttonOn]: this.props.showIds,
                [styles.buttonOff]: !this.props.showIds,
              })}
            >
              id's
            </button>{' '}
            <button
              onClick={() => this.props.onRefresh()}
              className={classNames(styles.button, styles.buttonOn)}
            >
              refresh
            </button>{' '}
          </span>
        }

        <button
          className={classNames(styles.button, styles.pin, {
            [styles.pinned]: this.props.pinned,
            [styles.buttonOn]: this.props.pinned,
            [styles.unpinned]: !this.props.pinned,
            [styles.buttonOff]: !this.props.pinned,
          })}
          onClick={() => this.props.onTogglePinned()}
        >
          {' '}
        </button>
      </div>
    );
  }
}
export namespace EditorPanel {
  export interface State {}
  export interface Props {
    searchEnabled: boolean;
    showIds: boolean;
    onTogglePinned: () => void;
    pinned: boolean;
    onRefresh: () => void;
    onToggleSearch: () => void;
    onShowIds: () => void;
    language: string;
    onChangeLanguage: (language: string) => void;
    languages: string[];
  }
}
