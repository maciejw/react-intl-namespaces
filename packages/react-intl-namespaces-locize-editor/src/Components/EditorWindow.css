.container {
  bottom: 0;
  box-shadow: -3px 0 5px 0 rgba(0, 0, 0, 0.5);
  position: fixed;
  right: 0;
  top: 0;
  z-index: 2000;
  display: none;
}
.containerVisible {
  display: block;
}

.iframe {
  border: none;
  height: 100%;
}
