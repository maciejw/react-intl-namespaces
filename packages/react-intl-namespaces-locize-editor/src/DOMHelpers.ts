function delay(timeout: number = 0) {
  return new Promise<void>(resolve => {
    window.setTimeout(resolve, timeout);
  });
}

export namespace DOMHelpers {
  export namespace Editor {
    export function toggleClass(
      element: HTMLElement,
      state: boolean,
      cssClass: string,
    ) {
      if (state) {
        element.classList.add(cssClass);
      } else {
        element.classList.remove(cssClass);
      }
    }
    export function createTargetElement(document: Document, id: string) {
      let editor = document.getElementById(id);
      if (editor == null) {
        editor = document.createElement('div');
        editor.id = id;
        document.body.appendChild(editor);
      }
      return editor;
    }
    export function isHtmlElement(element: EventTarget): element is HTMLElement;
    export function isHtmlElement(element: any): element is HTMLElement {
      return element instanceof HTMLElement;
    }

    export function findElementInDOMHierarchy(
      element: Element,
      filter: (htmlElement: HTMLElement) => boolean,
    ) {
      if (isHtmlElement(element)) {
        let e = element;
        while (e !== e.ownerDocument.body) {
          if (filter(e)) {
            return true;
          }
          if (e.parentElement === null) {
            return false;
          }
          e = e.parentElement;
        }
      }
      return false;
    }
    export function ignoreEditorOrTranslatedFilter(e: HTMLElement) {
      return (
        e.dataset.ignoreEditor === 'true' || e.dataset.translated === 'true'
      );
    }
    function isTranslatedOrIgnored(element: Element) {
      return findElementInDOMHierarchy(
        element,
        DOMHelpers.Editor.ignoreEditorOrTranslatedFilter,
      );
    }

    export function locateResourceContext(element: EventTarget) {
      if (isHtmlElement(element)) {
        const resourceContainer = element.parentElement;

        if (!resourceContainer || isTranslatedOrIgnored(resourceContainer)) {
          return false;
        }

        const {
          namespace,
          key,
          defaultMessage,
          description,
        } = resourceContainer.dataset;

        if (namespace && key) {
          return { namespace, key, defaultMessage, description };
        }
      }
      return false;
    }
  }

  export namespace EditorWindow {
    export function postMessage(
      window: Window | null,
      message: any,
      targetOrigin: string,
      transfer?: any[],
    ) {
      if (window !== null) {
        window.postMessage(message, targetOrigin, transfer);
      }
    }

    export function focus(window: Window | null) {
      if (window !== null) {
        window.focus();
      }
    }

    export async function openIfNecessary(
      window: Window,
      openedWindow: Window | null,
      windowOpenTimeout: number,
      url: string,
      target?: string,
      features?: string,
      replace?: boolean,
    ) {
      if (openedWindow === null || openedWindow.closed) {
        openedWindow = window.open(url, target, features, replace);
        await delay(windowOpenTimeout);
      }

      return openedWindow;
    }
  }
}
