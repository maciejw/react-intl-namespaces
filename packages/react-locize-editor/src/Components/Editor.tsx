import * as React from 'react';
import { LocizeEditorBinding } from '../locizeEditorBinding';

const editor = new LocizeEditorBinding({
  backend: {
    apiKey: '6b9ac145-b395-47dc-bcc4-001398c4c843',
    projectId: '2592abb8-1129-457d-a06b-836745d33c55',
    referenceLng: 'en',
  },
  enabled: true,
});

export class Editor extends React.Component {
  constructor(props: {}, context: {}) {
    super(props, context);
  }
  public render() {
    return null;
  }
}
