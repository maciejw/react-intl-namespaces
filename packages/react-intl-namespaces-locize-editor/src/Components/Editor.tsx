import * as React from 'react';
import * as ReactDOM from 'react-dom';

import { invariant } from '../invariant';
import { EditorPanel } from './EditorPanel';
import { EditorWindow } from './EditorWindow';

export class Editor extends React.Component<
  Editor.RequiredProps & Partial<Editor.OptionalProps>
> {
  public render() {
    return <EditorComponent {...Editor.defaultProps} {...this.props} />;
  }
}
class EditorComponent extends React.Component<
  Editor.RequiredProps & Editor.OptionalProps,
  Editor.State
> {
  private message: (ev: MessageEvent) => void;
  // prettier-ignore
  private keypress: (ev: KeyboardEvent) => void;
  // prettier-ignore
  private click: (e: MouseEvent) => void;
  // prettier-ignore
  private postMessage!: Promise<EditorWindow.PostMessage>;
  // prettier-ignore

  private editor: HTMLElement;

  constructor(props: Editor.RequiredProps & Editor.OptionalProps, context: {}) {
    super(props, context);
    this.state = {
      searchEnabled: true,
      showEditor: this.props.enabled,
      showIds: false,
    };

    let editor = document.getElementById('locize-editor');
    // invariant(editor === null, 'You should use only one Editor');

    editor = document.createElement('div');
    editor.id = 'locize-editor';
    document.body.appendChild(editor);
    this.editor = editor;

    this.click = (ev: MouseEvent) => this.lookupInEditor(ev);

    const { toggleKeyModifier, toggleKeyCode } = this.props;

    this.keypress = (ev: KeyboardEvent) => {
      if (ev[toggleKeyModifier] && ev.which === toggleKeyCode) {
        this.onSearchEnabled();
      }
    };
    this.message = (ev: MessageEvent) => {
      if (ev.data[toggleKeyModifier] && ev.data.which === toggleKeyCode) {
        this.onSearchEnabled();
      }
    };

    document.body.addEventListener('click', this.click);
    document.addEventListener('keypress', this.keypress);
    window.addEventListener('message', this.message);
  }

  public componentWillUnmount() {
    document.body.removeChild(this.editor);
    document.removeEventListener('keypress', this.keypress);
    window.removeEventListener('message', this.message);
  }

  public render() {
    const { searchEnabled, showIds } = this.state;
    return ReactDOM.createPortal(
      <div data-ignore-locize-editor="true">
        <EditorPanel
          showIds={showIds}
          searchEnabled={searchEnabled}
          onRefresh={() => this.onRefresh()}
          onSearchEnabled={() => this.onSearchEnabled()}
          onShowIds={() => this.onShowIds()}
          language={this.props.language}
          onChangeLanguage={this.props.onChangeLanguage}
          getLanguages={this.props.getLanguages}
        />
        <EditorWindow
          mode={this.props.mode}
          editorWidthInPixels={this.props.editorWidthInPixels}
          url={this.props.url}
          onOpen={i => this.open(i)}
        />
      </div>,
      this.editor,
    );
  }
  private onShowIds() {
    const { showIds: oldShowIds, ...rest } = this.state;
    const showIds = !oldShowIds;
    this.setState({ showIds, ...rest });
    this.props.onShowIds(showIds);
  }
  private open(callback: Promise<EditorWindow.PostMessage>) {
    this.postMessage = callback;
  }

  private isTranslatedOrIgnored(element: Element) {
    if (this.isHtmlElement(element)) {
      let e = element;
      while (e.tagName.toLowerCase() !== 'body') {
        if (
          e.dataset.ignoreLocizeEditor === 'true' ||
          e.dataset.translated === 'true'
        ) {
          return true;
        }
        if (e.parentElement === null) {
          return false;
        }
        e = e.parentElement;
      }
    }
    return false;
  }
  private isHtmlElement(element: Element): element is HTMLElement {
    return element instanceof HTMLElement;
  }

  private async lookupInEditor(e: MouseEvent) {
    if (!this.state.searchEnabled) {
      return;
    }
    e.preventDefault();

    if (e.srcElement) {
      const resourceContainer: HTMLElement | null = e.srcElement.parentElement;

      if (!resourceContainer || this.isTranslatedOrIgnored(resourceContainer)) {
        return;
      }

      const {
        ns,
        key,
        defaultMessage,
        description,
      } = resourceContainer.dataset;

      let message: Editor.SearchMessage;

      if (ns && key) {
        message = this.createMessage(ns, key, defaultMessage, description);

        const postMessage = await this.postMessage;

        if (this.props.mode === 'window') {
          if (!window.closed) {
            postMessage(message, this.props.url);
          }
        }
        if (this.props.mode === 'iframe') {
          postMessage(message, this.props.url);
        }
      } else {
        alert(
          `Missing key and namespace of resource, try search for a text or select ID mode`,
        );
      }
    }
  }
  private createMessage(
    ns: string,
    key: string,
    defaultMessage?: string,
    description?: string,
  ): Editor.SearchMessage {
    const { language: lng, projectId, version } = this.props;
    return {
      lng,
      message: 'searchForKey',
      ns,
      projectId,
      token: key,
      version,
    };
  }

  private onSearchEnabled() {
    const { searchEnabled } = this.state;
    this.setState({ searchEnabled: !searchEnabled });
  }
  private onRefresh() {
    this.props.onRefresh();
  }
}

export namespace Editor {
  export interface State {
    searchEnabled: boolean;
    showIds: boolean;
    showEditor: boolean;
  }
  export const defaultProps: Editor.OptionalProps = {
    editorWidthInPixels: 700,
    enabled: false,
    getLanguages: () => [],
    mode: 'iframe',
    onChangeLanguage: language => void 0,
    onRefresh: () => void 0,
    onShowIds: show => void 0,
    toggleKeyCode: 24,
    toggleKeyModifier: 'ctrlKey',
    url: 'https://www.locize.io',
    version: 'latest',
  };

  export interface OptionalProps {
    editorWidthInPixels: number;
    enabled: boolean;
    toggleKeyCode: number;
    toggleKeyModifier: 'ctrlKey' | 'altKey' | 'shiftKey';
    mode: 'window' | 'iframe';
    url: string;
    version: string;
    onShowIds: (show: boolean) => void;
    onChangeLanguage: (language: string) => void;
    getLanguages: () => string[];
    onRefresh: () => void;
  }

  export interface RequiredProps {
    apiKey: string;
    projectId: string;
    referenceLanguage: string;
    language: string;
  }
  export interface SearchMessage {
    message: 'searchForKey';
    projectId: string;
    version: string;
    lng: string;
    ns: string;
    token: string;
    defaultMessage?: string;
    description?: string;
  }
}
