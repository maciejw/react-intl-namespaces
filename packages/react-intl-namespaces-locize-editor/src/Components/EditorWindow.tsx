import * as React from 'react';
import * as ReactDOM from 'react-dom';

import * as styles from './EditorWindow.css';

function delay(timeout: number = 0) {
  return new Promise<void>(resolve => {
    window.setTimeout(resolve, timeout);
  });
}
export class EditorWindow extends React.Component<EditorWindow.Props> {
  // prettier-ignore
  private window: Window | undefined;
  constructor(props: EditorWindow.Props, state: {}) {
    super(props, state);
  }

  public componentDidMount() {
    if (this.props.mode === 'window') {
      const openedWindow = window.open(
        this.props.url,
        'locize-editor',
        '',
        true,
      );

      if (openedWindow !== null) {
        this.props.onOpen(
          new Promise(async resolve => {
            await delay(3000);
            resolve(openedWindow);
          }),
        );
      }
    }
  }

  public render() {
    switch (this.props.mode) {
      case 'iframe':
        const { container, iframe } = EditorWindow.inlineStyles(
          this.props.editorWidthInPixels,
        );
        return (
          <div
            className={styles.container}
            style={container}
            data-ignore-locize-editor="true"
          >
            <iframe
              className={styles.iframe}
              ref={e => {
                if (e !== null) {
                  this.props.onOpen(
                    new Promise(resolve => {
                      resolve(e.contentWindow);
                    }),
                  );
                }
              }}
              style={iframe}
              src={this.props.url}
            />
          </div>
        );
      case 'window':
        return null;
    }
  }
}
export namespace EditorWindow {
  export interface Props {
    url: string;
    onOpen: (window: Promise<Window>) => void;
    editorWidthInPixels: number;
    mode: 'iframe' | 'window';
  }

  export const inlineStyles: (
    editorWidthInPixels: number,
  ) => Record<
    'container' | 'iframe',
    React.CSSProperties
  > = editorWidthInPixels => ({
    container: {
      width: `${editorWidthInPixels}px`,
    },
    iframe: {
      width: `${editorWidthInPixels}px`,
    },
  });
}
