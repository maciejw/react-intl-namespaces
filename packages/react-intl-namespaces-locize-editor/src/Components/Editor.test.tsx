import { mount, ReactWrapper } from 'enzyme';
import * as React from 'react';
import {
  FormattedMessage,
  IntlBackendProvider,
  IntlNamespaceProvider,
} from 'react-intl-namespaces';

function delay(timeout: number = 0) {
  return new Promise<void>(resolve => {
    window.setTimeout(resolve, timeout);
  });
}

import { DOMHelpers } from '../DOMHelpers';
import { Editor, EditorComponent } from './Editor';
import { EditorPanel } from './EditorPanel';
import { EditorWindow } from './EditorWindow';

const app = document.createElement('div');
app.id = 'app';

const app1 = document.createElement('div');
app1.id = 'app';
const app2 = document.createElement('div');
app2.id = 'app';

const options: Editor.RequiredProps = {
  apiKey: '6b9ac145-b395-47dc-bcc4-001398c4c843',
  language: 'en',
  projectId: '2592abb8-1129-457d-a06b-836745d33c55',
  referenceLanguage: 'en',
};
describe('Editor', () => {
  function createMocks() {
    const BodyMock = jest.fn<HTMLBodyElement>(() => ({
      addEventListener: jest.fn((type, listener) =>
        document.body.addEventListener(type, listener),
      ),
      appendChild: jest.fn(newChild => document.body.appendChild(newChild)),
      removeChild: jest.fn(oldChild => document.body.removeChild(oldChild)),
      removeEventListener: jest.fn((type, ev, options) =>
        document.body.removeEventListener(type, ev, options),
      ),
    }));
    const bodyMock = new BodyMock();
    const DocumentMock = jest.fn<Document>(() => ({
      addEventListener: jest.fn((type, listener, options) =>
        document.addEventListener(type, listener, options),
      ),
      body: bodyMock,
      createElement: jest.fn(tagName => document.createElement(tagName)),
      getElementById: jest.fn(id => document.getElementById(id)),
      removeEventListener: jest.fn((type, ev, options) =>
        document.removeEventListener(type, ev, options),
      ),
    }));
    const documentMock = new DocumentMock();
    const WindowMock = jest.fn<Window>(() => ({
      addEventListener: jest.fn((type, listener, options) =>
        window.addEventListener(type, listener, options),
      ),
      document: documentMock,
      postMessage: jest.fn(),
      removeEventListener: jest.fn((type, ev, options) =>
        window.removeEventListener(type, ev, options),
      ),
    }));
    const windowMock = new WindowMock();

    return { windowMock, documentMock, bodyMock };
  }

  it('should render hidden by default', () => {
    const wrapper = mount(<Editor {...options} />);

    expect(wrapper.html()).toMatchSnapshot();

    wrapper.unmount();

    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should do nothing by default when refresh button is clicked', () => {
    const wrapper = mount(<Editor {...options} enabled={true} />);
    const panel: React.ComponentClass<EditorPanel.Props> = EditorPanel;
    const editorPanel: ReactWrapper<
      EditorPanel.Props,
      EditorPanel.State
    > = wrapper.find(EditorPanel);

    editorPanel.props().onRefresh();
  });
  it('should refresh when refresh button is clicked', () => {
    const onRefresh = jest.fn();
    const wrapper = mount(
      <Editor {...options} enabled={true} onRefresh={onRefresh} />,
    );
    const panel: React.ComponentClass<EditorPanel.Props> = EditorPanel;
    const editorPanel: ReactWrapper<
      EditorPanel.Props,
      EditorPanel.State
    > = wrapper.find(EditorPanel);

    editorPanel.props().onRefresh();

    expect(onRefresh).toMatchSnapshot();
  });

  it('should do nothing by default when show ids button is clicked', () => {
    const wrapper = mount(<Editor {...options} enabled={true} />);
    const panel: React.ComponentClass<EditorPanel.Props> = EditorPanel;
    const editorPanel: ReactWrapper<
      EditorPanel.Props,
      EditorPanel.State
    > = wrapper.find(EditorPanel);

    editorPanel.props().onShowIds();
  });
  it('should show ids when show ids button is clicked', () => {
    const onShowIds = jest.fn();
    const wrapper = mount(
      <Editor {...options} enabled={true} onShowIds={onShowIds} />,
    );
    const panel: React.ComponentClass<EditorPanel.Props> = EditorPanel;
    const editorPanel: ReactWrapper<
      EditorPanel.Props,
      EditorPanel.State
    > = wrapper.find(EditorPanel);

    editorPanel.props().onShowIds();

    expect(onShowIds).toMatchSnapshot();

    editorPanel.props().onShowIds();

    expect(onShowIds).toMatchSnapshot();
  });
  it('should do unpin when pin button is clicked', () => {
    const wrapper = mount(<Editor {...options} enabled={true} />);
    const panel: React.ComponentClass<EditorPanel.Props> = EditorPanel;
    const editorPanel: ReactWrapper<
      EditorPanel.Props,
      EditorPanel.State
    > = wrapper.find(EditorPanel);

    const editorComponent: ReactWrapper<{}, Editor.State> = wrapper.find(
      EditorComponent,
    );

    expect(editorComponent.instance().state.pinned).toBe(true);

    editorPanel.props().onTogglePinned();

    expect(editorComponent.instance().state.pinned).toBe(false);
  });
  it('should do nothing by default when new language is selected', () => {
    const wrapper = mount(<Editor {...options} enabled={true} />);
    const panel: React.ComponentClass<EditorPanel.Props> = EditorPanel;
    const editorPanel: ReactWrapper<
      EditorPanel.Props,
      EditorPanel.State
    > = wrapper.find(EditorPanel);

    editorPanel.props().onChangeLanguage('en');
  });

  it('should change language when new language is selected', () => {
    const onChangeLanguage = jest.fn();
    const wrapper = mount(
      <Editor
        {...options}
        enabled={true}
        onChangeLanguage={onChangeLanguage}
      />,
    );
    const panel: React.ComponentClass<EditorPanel.Props> = EditorPanel;
    const editorPanel: ReactWrapper<
      EditorPanel.Props,
      EditorPanel.State
    > = wrapper.find(EditorPanel);

    editorPanel.props().onChangeLanguage('en');

    expect(onChangeLanguage).toMatchSnapshot();
  });

  beforeEach(() => {
    document.body.appendChild(app);
    document.body.appendChild(app1);
    document.body.appendChild(app2);
  });

  afterEach(() => {
    document.body.removeChild(app);
    document.body.removeChild(app1);
    document.body.removeChild(app2);
  });

  it('should search for resource when clicked formatted message text', async () => {
    const { documentMock, windowMock } = createMocks();
    const wrapper = mount(
      <div>
        <IntlBackendProvider locale="en" includeMetadata={true}>
          <IntlNamespaceProvider namespace="namespace">
            <FormattedMessage
              id="message"
              defaultMessage="default message"
              description="description"
            />
          </IntlNamespaceProvider>
        </IntlBackendProvider>
        <Editor
          {...options}
          editorId="editor1"
          enabled={true}
          window={windowMock}
          document={documentMock}
        />
      </div>,
      { attachTo: app1 },
    );

    const spanWithResourceText = wrapper
      .find(FormattedMessage)
      .find('span>span');

    const editorWindowWrapper: ReactWrapper<EditorWindow.Props> = wrapper.find(
      EditorWindow,
    );
    editorWindowWrapper.props().onOpen(Promise.resolve(windowMock.postMessage));

    const element = spanWithResourceText.getDOMNode();

    if (DOMHelpers.Editor.isHtmlElement(element)) {
      element.click();
    }

    const panel = wrapper.find(EditorPanel).getDOMNode();

    if (DOMHelpers.Editor.isHtmlElement(panel)) {
      panel.click();
    }
    await delay(100);
    wrapper.unmount();
    expect(windowMock).toMatchSnapshot();
  });

  it('should do not search for resource after search is disabled', async () => {
    const { documentMock, windowMock } = createMocks();
    const wrapper = mount(
      <div>
        <IntlBackendProvider locale="en" includeMetadata={true}>
          <IntlNamespaceProvider namespace="namespace">
            <FormattedMessage
              id="message"
              defaultMessage="default message"
              description="description"
            />
          </IntlNamespaceProvider>
        </IntlBackendProvider>
        <Editor
          {...options}
          enabled={true}
          editorId="editor2"
          window={windowMock}
          document={documentMock}
        />
      </div>,
      { attachTo: app2 },
    );
    const panel: React.ComponentClass<EditorPanel.Props> = EditorPanel;
    const editorPanel: ReactWrapper<
      EditorPanel.Props,
      EditorPanel.State
    > = wrapper.find(EditorPanel);

    editorPanel.props().onToggleSearch();

    const spanWithResourceText = wrapper
      .find(FormattedMessage)
      .find('span>span');

    const element = spanWithResourceText.getDOMNode();

    if (DOMHelpers.Editor.isHtmlElement(element)) {
      element.click();
    }

    await delay(100);
    wrapper.unmount();
    expect(windowMock).toMatchSnapshot();
  });

  it('should switch search enabled on key pressed', async () => {
    const { documentMock, windowMock } = createMocks();
    const wrapper = mount(
      <Editor
        {...options}
        enabled={true}
        window={windowMock}
        document={documentMock}
      />,
    );
    const editorComponent: ReactWrapper<
      Editor.RequiredProps,
      Editor.State
    > = wrapper.find(EditorComponent);

    expect(editorComponent.instance().state.searchEnabled).toBe(true);

    const keypressOptions = {
      ctrlKey: true,
      key: 'Enter',
    };
    const keypress = new KeyboardEvent('keypress', keypressOptions);

    document.dispatchEvent(keypress);
    await delay();
    expect(editorComponent.instance().state.searchEnabled).toBe(false);
  });

  it('should switch search enabled on message post', async () => {
    const { documentMock, windowMock } = createMocks();
    const wrapper = mount(
      <Editor
        {...options}
        enabled={true}
        window={windowMock}
        document={documentMock}
      />,
    );
    const editorComponent: ReactWrapper<
      Editor.RequiredProps,
      Editor.State
    > = wrapper.find(EditorComponent);

    expect(editorComponent.instance().state.searchEnabled).toBe(true);

    const keypressOptions = {
      ctrlKey: true,
      key: 'Enter',
    };
    const message = new MessageEvent('message', { data: keypressOptions });

    window.dispatchEvent(message);
    await delay();
    expect(editorComponent.instance().state.searchEnabled).toBe(false);

    wrapper.unmount();
  });
});
