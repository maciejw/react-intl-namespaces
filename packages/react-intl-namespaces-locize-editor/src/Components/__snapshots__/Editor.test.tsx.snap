// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor should change language when new language is selected 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "en",
    ],
  ],
}
`;

exports[`Editor should do not search for resource after search is disabled 1`] = `
Object {
  "addEventListener": [MockFunction] {
    "calls": Array [
      Array [
        "message",
        [Function],
      ],
    ],
  },
  "document": Object {
    "addEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "keypress",
          [Function],
        ],
      ],
    },
    "body": Object {
      "addEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [Function],
          ],
        ],
      },
      "appendChild": [MockFunction] {
        "calls": Array [
          Array [
            <div
              id="editor2"
            />,
          ],
        ],
      },
      "removeChild": [MockFunction] {
        "calls": Array [
          Array [
            <div
              id="editor2"
            />,
          ],
        ],
      },
      "removeEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [Function],
          ],
        ],
      },
    },
    "createElement": [MockFunction] {
      "calls": Array [
        Array [
          "div",
        ],
      ],
    },
    "getElementById": [MockFunction] {
      "calls": Array [
        Array [
          "editor2",
        ],
      ],
    },
    "removeEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "keypress",
          [Function],
        ],
      ],
    },
  },
  "postMessage": [MockFunction],
  "removeEventListener": [MockFunction] {
    "calls": Array [
      Array [
        "message",
        [Function],
      ],
    ],
  },
}
`;

exports[`Editor should refresh when refresh button is clicked 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
}
`;

exports[`Editor should render hidden by default 1`] = `"<div data-ignore-editor=\\"true\\"><div class=\\"panel pinned\\"><span class=\\"content\\"><div class=\\"title\\">locize editor</div> <button class=\\"button buttonOn\\">On</button> <button class=\\"button buttonOff\\">id's</button> <button class=\\"button buttonOn\\">refresh</button> </span><button class=\\"button pin pinned buttonOn\\"> </button></div><div class=\\"container containerVisible\\" style=\\"width: 700px;\\"><iframe class=\\"iframe\\" style=\\"width: 700px;\\" src=\\"https://www.locize.io\\"></iframe></div></div>"`;

exports[`Editor should render hidden by default 2`] = `null`;

exports[`Editor should search for resource when clicked formatted message text 1`] = `
Object {
  "addEventListener": [MockFunction] {
    "calls": Array [
      Array [
        "message",
        [Function],
      ],
    ],
  },
  "document": Object {
    "addEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "keypress",
          [Function],
        ],
      ],
    },
    "body": Object {
      "addEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [Function],
          ],
        ],
      },
      "appendChild": [MockFunction] {
        "calls": Array [
          Array [
            <div
              id="editor1"
            />,
          ],
        ],
      },
      "removeChild": [MockFunction] {
        "calls": Array [
          Array [
            <div
              id="editor1"
            />,
          ],
        ],
      },
      "removeEventListener": [MockFunction] {
        "calls": Array [
          Array [
            "click",
            [Function],
          ],
        ],
      },
    },
    "createElement": [MockFunction] {
      "calls": Array [
        Array [
          "div",
        ],
      ],
    },
    "getElementById": [MockFunction] {
      "calls": Array [
        Array [
          "editor1",
        ],
      ],
    },
    "removeEventListener": [MockFunction] {
      "calls": Array [
        Array [
          "keypress",
          [Function],
        ],
      ],
    },
  },
  "postMessage": [MockFunction] {
    "calls": Array [
      Array [
        Object {
          "lng": "en",
          "message": "searchForKey",
          "ns": "namespace",
          "projectId": "2592abb8-1129-457d-a06b-836745d33c55",
          "token": "message",
          "version": "latest",
        },
        "https://www.locize.io",
      ],
    ],
  },
  "removeEventListener": [MockFunction] {
    "calls": Array [
      Array [
        "message",
        [Function],
      ],
    ],
  },
}
`;

exports[`Editor should show ids when show ids button is clicked 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      true,
    ],
  ],
}
`;

exports[`Editor should show ids when show ids button is clicked 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      true,
    ],
    Array [
      false,
    ],
  ],
}
`;
